pkg/distributor/distributor.go

```diff
@@ -965,15 +991,21 @@ func (d *Distributor) prePushValidationMiddleware(next push.Func) push.Func {
 			req.Metadata = util.RemoveSliceIndexes(req.Metadata, removeIndexes)
 		}
 
-		if validatedSamples == 0 && validatedMetadata == 0 {
+		if validatedSamples == 0 && validatedHistograms == 0 && validatedMetadata == 0 {
 			return &mimirpb.WriteResponse{}, firstPartialErr
 		}
 
-		totalN := validatedSamples + validatedExemplars + validatedMetadata
+		totalN := validatedSamples + validatedExemplars + validatedHistograms + validatedMetadata
 		if !d.ingestionRateLimiter.AllowN(now, userID, totalN) {
-			d.discardedSamplesRateLimited.WithLabelValues(userID, group).Add(float64(validatedSamples))
-			d.discardedExemplarsRateLimited.WithLabelValues(userID).Add(float64(validatedExemplars))
-			d.discardedMetadataRateLimited.WithLabelValues(userID).Add(float64(validatedMetadata))
+			if validatedSamples+validatedHistograms > 0 {
+				d.discardedSamplesRateLimited.WithLabelValues(userID, group).Add(float64(validatedSamples + validatedHistograms))
+			}
+			if validatedExemplars > 0 {
+				d.discardedExemplarsRateLimited.WithLabelValues(userID).Add(float64(validatedExemplars))
+			}
+			if validatedMetadata > 0 {
+				d.discardedMetadataRateLimited.WithLabelValues(userID).Add(float64(validatedMetadata))
+			}
 			// Return a 429 here to tell the client it is going too fast.
```

Comment:
Can we please do optimizations in separate PR, with benchmarks?

Reply:
I was told .WithLabelValues is costly, certainly more costly than an `if` . What do you expect from a benchmark?

Reply:
The point is that this is an optimization compared to the previous state on `main` branch. My question is: is it worth optimizing for and paying the price of extra code? Did we measure the impact of not doing that?

Reply:
discussed IRL and decided to revert this change as calling the function per request is ok (as opposed to per timeseries or sample)